test/ability_snow_cloak.c

```diff
@@ -0,0 +1,29 @@
+#include "global.h"
+#include "test_battle.h"
+
+SINGLE_BATTLE_TEST("Snow Cloak prevents damage from hail")
+{
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_GLACEON) { Ability(ABILITY_SNOW_CLOAK); };
+    } WHEN {
+        TURN { MOVE(player, MOVE_HAIL); MOVE(opponent, MOVE_SKILL_SWAP); }
+    } SCENE {
+        NONE_OF { HP_BAR(player); }
+    }
+}
+
+SINGLE_BATTLE_TEST("Snow Cloak reduces accuracy during hail")
```

Comment:
```suggestion
SINGLE_BATTLE_TEST("Snow Cloak increases evasion during hail")
```

Reply:
I wrote it like that to be consistent with the Sand Veil test, but imo they should both be `"X increases evasion during Y"` so I've made that change for both.